src/trans/gpu/algor/hicblas_cutlass.cuda.h

```diff
@@ -154,11 +154,11 @@ class cutlass_sgemm_grouped<CutlassType::cutlass_fp32, TransA, TransB> {
 
 } // namespace detail
 template <cublasOperation_t TransA, cublasOperation_t TransB>
-void cutlass_sgemm_wrapper_grouped_op(int resol_id, int blas_id, int m, int *n, int *k,
+void cutlass_sgemm_wrapper_grouped_op(int resol_id, int blas_id, int m, const int *n, const int *k,
                                       float alpha, const float *A, int lda,
-                                      int64_t *offsetsA, const float *B, int ldb,
-                                      int64_t *offsetsB, float beta, float *C,
-                                      int ldc, int64_t *offsetsC, int batchCount,
+                                      const int64_t *offsetsA, const float *B, const int *ldb,
```

Comment:
More generally, what's the rationale for adding the `const` decorator to the integer pointer arguments?

Reply:
to make sure we cannot write into the data that we are pointing to, and also just consistency. It is natural that offsetsA cannot be modified, so I added it to some inner function, but that certainly has to be propagated outwards up to the entry function. Note that scalars are somewhat different - one could also add const for those, but modifying them does not propagate, so it is less harmful. I hope that makes sense.

Reply:
It does! Many thanks. I am still very much a C++ n00b.